src/webapi/event.rs

```diff
@@ -10,26 +12,169 @@ use webcore::try_from::{TryFrom, TryInto};
+    }
+
+    /// A historical alias to `Event.stopPropagation()`.
+    /// Setting this to `true` before returning from an event handler will stop propagation
+    /// of the event.
+    ///
+    /// [(JavaScript docs)](https://developer.mozilla.org/en-US/docs/Web/API/Event/cancelBubble)
+    #[inline]
+    fn set_cancel_bubble( &self, value: bool ) {
+        js! { @(no_return)
+            @{self.as_ref()}.cancelBubble = @{value};
+        }
+    }
+
+    /// Indicates whether the event is cancelable.
+    ///
+    /// [(JavaScript docs)](https://developer.mozilla.org/en-US/docs/Web/API/Event/cancelable)
+    #[inline]
+    fn cancelable( &self ) -> bool {
+        js!(
+            return @{self.as_ref()}.cancelable;
+        ).try_into().unwrap()
+    }
+
+    /// A reference to the currently registered target of this event.
+    ///
+    /// [(JavaScript docs)](https://developer.mozilla.org/en-US/docs/Web/API/Event/currentTarget)
+    #[inline]
+    fn current_target( &self ) -> Option< Reference > {
+        js!(
+            return @{self.as_ref()}.currentTarget;
+        ).try_into().ok()
+    }
+
+    /// Indicates whether `preventDefault` has been called on this event.
+    ///
+    /// [(JavaScript docs)](https://developer.mozilla.org/en-US/docs/Web/API/Event/defaultPrevented)
+    #[inline]
+    fn default_prevented( &self ) -> bool {
+        js!(
+            return @{self.as_ref()}.defaultPrevented;
+        ).try_into().unwrap()
+    }
+
+    /// Indicates which phase of event flow is currently being evaluated.
+    ///
+    /// [(JavaScript docs)](https://developer.mozilla.org/en-US/docs/Web/API/Event/eventPhase)
+    fn event_phase( &self ) -> EventPhase {
+        match js!(
+            return @{self.as_ref()}.eventPhase;
+        ).try_into().unwrap() {
+            0 => EventPhase::None,
+            1 => EventPhase::Capturing,
+            2 => EventPhase::AtTarget,
+            3 => EventPhase::Bubbling,
+            _ => unreachable!("Unexpected EventPhase type"),
+        }
+    }
+
+    /// Prevents any further listeners from being called for this event.
+    ///
+    /// [(JavaScript docs)](https://developer.mozilla.org/en-US/docs/Web/API/Event/stopImmediatePropagation)
+    #[inline]
+    fn stop_immediate_propagation( &self ) {
+        js! { @(no_return)
+            @{self.as_ref()}.stopImmediatePropagation();
+        }
+    }
+
+    /// Stops the propagation of this event to descendants in the DOM.
+    ///
+    /// [(JavaScript docs)](https://developer.mozilla.org/en-US/docs/Web/API/Event/stopPropagation)
+    #[inline]
+    fn stop_propagation( &self ) {
+        js! { @(no_return)
+            @{self.as_ref()}.stopPropagation();
+        }
+    }
+
+
+    /// Returns a reference to the target to which this event was originally registered.
+    ///
+    /// [(JavaScript docs)](https://developer.mozilla.org/en-US/docs/Web/API/Event/target)
+    #[inline]
+    fn target( &self ) -> Option< Reference > {
```

Comment:
Hmm... it probably would make more sense to return an `EventTarget` instead of a `Reference` here. (Since you can only register events on `EventTarget`s.)

Looking at the [spec](https://dom.spec.whatwg.org/#concept-event) I can find the following:

```
An event has [...] target (null or an EventTarget object),
```

so if the spec is to be believed it looks like it indeed should be an `EventTarget`.